R/executionSettings.R

```diff
@@ -0,0 +1,167 @@
+
+# ExecutionSettings ----
+
+#' @description
+#' An R6 class to define an ExecutionSettings object
+#'
+#' @export
+ExecutionSettings <- R6::R6Class(
+  classname = "ExecutionSettings",
+  public = list(
+    initialize = function(connectionDetails = NULL,
+                          connection = NULL,
+                          cdmDatabaseSchema = NULL,
+                          workDatabaseSchema = NULL,
+                          tempEmulationSchema = NULL,
+                          targetCohortTable = NULL,
+                          cdmSourceName = NULL) {
+      stopifnot(is.null(connectionDetails) || is.null(connection))
+      .setClass(private = private, key = "connectionDetails", value = connectionDetails, class = "ConnectionDetails")
+      .setClass(private = private, key = ".connection", value = connection,
+                class = "DatabaseConnectorJdbcConnection", nullable = TRUE)
+      .setString(private = private, key = ".cdmDatabaseSchema", value = cdmDatabaseSchema)
+      .setString(private = private, key = ".workDatabaseSchema", value = workDatabaseSchema)
+      .setString(private = private, key = ".tempEmulationSchema", value = tempEmulationSchema)
+      .setString(private = private, key = ".targetCohortTable", value = targetCohortTable)
+      .setString(private = private, key = ".cdmSourceName", value = cdmSourceName)
+    },
+    
+    getDbms = function() {
+      dbms <- private$connectionDetails$dbms
+      return(dbms)
+    },
+    
+    # connect to database
+    connect = function() {
+      
+      # check if private$connection is NULL
+      conObj <- private$.connection
+      if (is.null(conObj)) {
+        private$.connection <- DatabaseConnector::connect(private$connectionDetails)
+      } else{
+        cli::cat_bullet(
+          "Connection object already open",
+          bullet = "info",
+          bullet_col = "blue"
+        )
+      }
+    },
+    
+    # disconnect to database
+    disconnect = function() {
```

Comment:
perhaps a "withConnection" withr style context can be implemented as a method see here:

https://github.com/OHDSI/DatabaseConnector/issues/269